app/components/base_component.rb

```diff
@@ -1,18 +1,17 @@
 class BaseComponent < ViewComponent::Base
-  def render_in(view_context, &block)
-    render_scripts_in(view_context)
-    super(view_context, &block)
-  end
-
-  def render_scripts_in(view_context)
+  def before_render
     return if @rendered_scripts
     @rendered_scripts = true
-    if view_context.respond_to?(:render_component_script) && self.class.scripts.present?
-      view_context.render_component_script(*self.class.scripts)
+    if helpers.respond_to?(:render_component_script) && self.class.scripts.present?
+      helpers.render_component_script(*self.class.scripts)
```

Comment:
- With nested components, the `view_context` is the parent component, not the base ActionView where helpers exist.
- `before_render` avoids needing to override `render_in`, and seems to exist for the sort of purpose we're using it for here (hooking).
- The library upgrade is necessary due to encountering the bug fixed in [2.42.0](https://viewcomponent.org/CHANGELOG.html#2420) ("Fix bug where `helpers` would instantiate and use a new `view_context` in each component.")

Reply:
These changes were addressed separately in #5627